src/embit/bip47.py

```diff
@@ -0,0 +1,258 @@
+    raw_blinded_payload = blinding_function(a, B, utxo_outpoint=input_utxo_outpoint, payload=payment_code)
+    return hexlify(raw_blinded_payload).decode()
+
+
+def get_payment_code_from_notification_tx(tx: Transaction, recipient_root: HDKey, coin: int = 0, account: int = 0, network=NETWORKS["main"]) -> str:
+    """If the tx is a BIP-47 notification tx for the recipient,
+        return the new payer's embedded payment_code, else None"""
+    # Notification txs have one output sent to the recipient's notification addr
+    # and another containing the payer's payment code in an OP_RETURN payload.
+    if len(tx.vout) < 2:
+        return False
+    
+    recipient_payment_code = get_payment_code(recipient_root, coin, account)
+    
+    matches_notification_addr = False
+    payload = None
+    for vout in tx.vout:
+        # Notification txs include a dust payment to the recipient's notification address
+        if vout.script_pubkey.script_type() is not None and vout.script_pubkey.address() == get_notification_address(recipient_payment_code, network=network):
+            matches_notification_addr = True
+            continue
+
+        # Payer's payment code will be in an OP_RETURN w/exactly 80 bytes of data
+        data = vout.script_pubkey.data
+        if data is not None and len(data) == 83 and data[0] == OPCODES.OP_RETURN and data[1] == OPCODES.OP_PUSHDATA1 and data[2] == 80:
+            # data = OP_RETURN OP_PUSHDATA1 (len of data) <data>
+            payload = data[3:]
+
+            if payload[0] != 1:
+                # Only version 1 currently supported
+                payload = None
+            continue
+
+    if not matches_notification_addr or payload is None:
+        return None
+    
+    # Bob selects the designated pubkey ("A")
+    # (the first tx input that exposes a pubkey in scriptsig or witness)
+    for vin in tx.vin:
+        if not vin.is_segwit:
+            # data = (1byte len of sig) <sig> (1byte len of pubkey) <pubkey>
+            sig_len = vin.script_sig.data[0]
+            A = ec.PublicKey.from_string(hexlify(vin.script_sig.data[sig_len + 2:]))
+            break
+
+        else:
+            # Witness should have [sig, pubkey]
+            A = ec.PublicKey.from_string(hexlify(vin.witness.items[1]))
+            break
+
+    if not A or len(A.serialize()) != 33:
+        return None
+    
+    # Bob selects the private key associated with his notification address (0th child)
+    recipient_notification_node = recipient_root.derive(f"m/47'/{coin}'/{account}'/0")
+    b = recipient_notification_node.secret
+
+    utxo_outpoint = vin.to_string()[:72]  # TODO: Is there a better way to get the outpoint?
+
+    # Unblind the payload using the reversible `blinding_function`.
+    raw_unblinded_payload = blinding_function(b, A, utxo_outpoint=utxo_outpoint, payload=payload)
+    return base58.encode_check(b'\x47' + raw_unblinded_payload)
+
+
+"""
+    TODO: Method to create notification transaction, etc.
+"""
```

Comment:
Are you planning to implement it in this PR or in another one?

Reply:
Either way. We can hold this PR open until a bit more is implemented or can supplement it with subsequent PRs. But I think what is here so far is a reasonable self-contained first step.

Reply:
I added a test that demonstrates how the notification tx can be constructed, but I don't see a need to add a method here to do that since it's just adding an `OP_RETURN` output to a psbt. If anything, that kind of generalized method would belong in `PSBT` but I don't think there's a strong need to add it.